CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -733,6 +733,107 @@ public CommandBase RefreshJobs
+                                "Confirm Delete",
+                                MessageBoxButton.YesNo);
+
+                            if (result == MessageBoxResult.Yes)
+                            {
+                                try
+                                {
+                                    this.LeftSpinnerIsVisible = true;
+                                    FirePropertyChangedEvent("LeftSpinnerIsVisible");
+
+                                    foreach (var job in this.jobs)
+                                    {
+                                        if (job.IsChecked)
+                                        {
+                                            await job.DeleteAsync(false);
+                                        }
+                                    }
+
+                                    //This is very hacky as the tab count never updated correctly if we refresh
+                                    //just after issueing delete async inspite of awaiting for result. Probably it is by default nature
+                                    //So keep on refreshing until we see the deleted record inside the refresh list
+                                    bool requiredRefresh = true;
+                                    while (requiredRefresh)
+                                    {
+                                        var refreshedJobs = await dataProvider.GetJobCollectionAsync();
+                                        requiredRefresh = refreshedJobs.Any(a => deletedIds.Contains(a.Id));
+                                        if (!requiredRefresh)
+                                        {
+                                            this.jobs = new ObservableCollection<JobModel>(refreshedJobs);
+                                            this.jobCollection = CollectionViewSource.GetDefaultView(this.jobs);
+                                            this.selectedJob = null;
+                                            this.Jobs.Refresh();
+
+                                            //Refresh the job pane
+                                            FirePropertyChangedEvent("Jobs");
+                                            FirePropertyChangedEvent("JobTabTitle");
+
+                                            //Refresh the task pane
+                                            FirePropertyChangedEvent("SelectedJob");
+                                            FirePropertyChangedEvent("TasksTabTitle");
+                                        }
+                                        else
+                                        {
+                                            await Task.Delay(1000);
+                                        }
+                                    }
+                                }
+                                catch(Exception ex)
+                                {
+                                    MessageBox.Show(ex.Message);

```

Comment:
I thought we tried to use Messenger instead of MessageBox or other hard-coed dialog references to avoid having modals in the ViewModel code?